src/dash/DashAdapter.js

```diff
@@ -718,6 +755,129 @@ function DashAdapter() {
         currentMediaInfo = {};
     }
 
+    /**
+     * Checks if the supplied manifest is compatible for application of the supplied patch
+     * @param  {object}  manifest
+     * @param  {object}  patch
+     * @return {boolean}
+     */
+    function isPatchValid(manifest, patch) {
+        let manifestId = dashManifestModel.getId(manifest);
+        let patchManifestId = patchManifestModel.getMpdId(patch);
+        let publishTime = dashManifestModel.getPublishTime(manifest);
+        let originalPublishTime = patchManifestModel.getOriginalPublishTime(patch);
+
+        // Patches are considered compatible if the following are true
+        // - MPD@id == Patch@mpdId
+        // - MPD@publishTime == Patch@originalPublishTime
+        // - All values in comparison exist
+        return !!(manifestId && patchManifestId && (manifestId == patchManifestId) &&
```

Comment:
Pulled in `Patch@publishTime > MPD@publishTime` check, clarified variable naming for it as well.

The patch modifying `MPD@publishTime` is a bit trickier, while generally a single operation should cause the change, technically a cascading set of changes could cause it. To avoid any heuristics that may be wrong or overly complicated, I enforced the check at patch application time, see below.